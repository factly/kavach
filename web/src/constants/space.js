//Actions
export const ADD_SPACE = 'ADD_SPACE';
export const ADD_SPACES = 'ADD_SPACES';
export const SET_SPACES_LOADING = 'SET_SPACES_LOADING';
export const RESET_SPACES = 'RESET_SPACES';
export const STOP_SPACES_LOADING = 'STOP_SPACES_LOADING';
export const SET_SELECTED_APP = 'SET_SELECTED_APP';
export const ADD_SPACE_ROLE_IDS = 'ADD_SPACE_ROLE_IDS';
export const ADD_SPACE_POLICY_IDS = 'ADD_SPACE_POLICY_IDS';
//API
export const SPACES_API = '/spaces';
