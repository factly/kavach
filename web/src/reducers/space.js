import {
  ADD_SPACE,
  SET_SPACES_LOADING,
  RESET_SPACES,
  STOP_SPACES_LOADING,
  SET_SELECTED_APP,
} from '../constants/space';

const initialState = {
  ids: [],
  details: {},
  selected: null,
  loading: true,
};

export default function spaces(state = initialState, action = {}) {
  switch (action.type) {
    case SET_SPACES_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case STOP_SPACES_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case ADD_SPACE:
      return {
        ...state,
        ids: action.payload.map((item) => item.id),
        details: action.payload.reduce((obj, item) => Object.assign(obj, { [item.id]: item }), {}),
      };
    case RESET_SPACES:
      return {
        ...state,
        ids: [],
        details: {},
        loading: true,
      };
    case SET_SELECTED_APP:
      return {
        ...state,
        selected: action.payload,
      };
    default:
      return state;
  }
}
