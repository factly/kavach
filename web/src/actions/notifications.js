import { ADD_NOTIFICATION } from '../constants/notifications';

export const addErrorNotification = (data) => ({
  type: ADD_NOTIFICATION,
  payload: {
    type: 'error',
    title: 'Error',
    message: data,
  },
});

export const addSuccessNotification = (data) => ({
  type: ADD_NOTIFICATION,
  payload: {
    type: 'success',
    title: 'Success',
    message: data,
  },
});
