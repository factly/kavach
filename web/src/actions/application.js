import axios from 'axios';
import {
  ADD_APPLICATION,
  ADD_APPLICATIONS,
  ADD_APPLICATIONS_REQUEST,
  SET_APPLICATIONS_LOADING,
  RESET_APPLICATIONS,
  APPLICATIONS_API,
  ADD_SPACE_IDS,
  SET_DEFAULT_APPLICATION_LOADING,
  GET_DEFAULT_APPLICATIONS,
} from '../constants/application';
import { ADD_APPLICATION_IDS, ORGANISATIONS_API } from '../constants/organisations';
import { buildObjectOfItems, deleteKeys, getIds, getValues } from '../utils/objects';
import { addMedia, addMediaList } from './media';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { addSpaces } from './space';
import { addApplicationTokens } from './token';
import { addUsersList } from './users';
export const getApplications = () => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications')
      .then((response) => {
        dispatch(addApplicationList(response.data));
        dispatch(addApplicationIds(getIds(response.data)));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopApplicationLoading());
      });
  };
};

export const addDefaultApplication = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .post(
        ORGANISATIONS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/default',
      )
      .then(() => {
        dispatch(addSuccessNotification('Application Added Successfully'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopApplicationLoading());
      });
  };
};

export const removeDefaultApplication = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .delete(
        ORGANISATIONS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/default',
      )
      .then(() => {
        dispatch(addSuccessNotification('Application removed succesfully'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopApplicationLoading());
      });
  };
};

export const getDefaultApplications = () => {
  return (dispatch, getState) => {
    dispatch(setDefaultApplicationsLoading());
    return axios
      .get(ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications/default')
      .then((response) => {
        dispatch(addDefaultApplicationsList(response.data));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopDefaultApplicationLoading());
      });
  };
};

export const getApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(APPLICATIONS_API + '/' + getState().organisations.selected + '/applications/' + id)
      .then((response) => {
        if (response.data.medium_id) {
          dispatch(addMedia(response.data.medium));
        }
        deleteKeys([response.data], ['medium']);
        dispatch(addUsersList(response.data.users));
        response.data.users = getIds(response.data.users);
        response.data.spaces = getIds(response.data.spaces)
        response.data.tokens = getIds(response.data.tokens)
        const spaces = getValues([response.data], 'spaces');
        // dispatch(addSpaces(spaces));
        dispatch(addApplication(response.data));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopApplicationLoading());
      });
  };
};

export const createApplication = (data) => {
  data.is_default = false;
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .post(ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications', data)
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application Added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateApplication = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .put(
        ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications/' + data.id,
        data,
      )
      .then(() => {
        dispatch(addSuccessNotification('Application Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopApplicationLoading());
      });
  };
};

export const deleteApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .delete(ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications/' + id)
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addApplications = (applications) => {
  return (dispatch) => {
    dispatch(
      addMediaList(
        applications
          .filter((application) => application.medium)
          .map((application) => application.medium),
      ),
    );
    dispatch(
      addApplicationList(
        applications.map((application) => {
          return { ...application, medium: application.medium?.id };
        }),
      ),
    );
  };
};

export const loadingApplications = () => ({
  type: SET_APPLICATIONS_LOADING,
  payload: true,
});

export const stopApplicationLoading = () => ({
  type: SET_APPLICATIONS_LOADING,
  payload: false,
});

export const getApplicationByID = (data) => ({
  type: ADD_APPLICATION,
  payload: data,
});

export const addApplicationList = (data) => (dispatch) => {
  dispatch(loadingApplications());
  const medium = getValues(data, 'medium');
  dispatch(addMediaList(medium));
  deleteKeys(data, ['medium']);
  const spaces = getValues(data, 'spaces');
  dispatch(addSpaces(spaces));

  data.forEach((application) => {
    application.spaces = getIds(application.spaces);
    application.users = getIds(application.users);
    dispatch(addApplicationTokens(application.id, application.tokens));
    application.tokens = getIds(application.tokens);
  });

  dispatch({
    type: ADD_APPLICATIONS,
    payload: buildObjectOfItems(data),
  });
  dispatch(stopApplicationLoading());
};

export const addApplicationsRequest = (data) => ({
  type: ADD_APPLICATIONS_REQUEST,
  payload: data,
});

export const resetApplications = () => ({
  type: RESET_APPLICATIONS,
});

export const addApplicationIds = (data) => ({
  type: ADD_APPLICATION_IDS,
  payload: data,
});

export const addApplication = (data) => ({
  type: ADD_APPLICATION,
  payload: data,
});

export const addSpaceIDs = (appID, data) => ({
  type: ADD_SPACE_IDS,
  payload: {
    appID: appID,
    data: data,
  },
});

export const setDefaultApplicationsLoading = () => ({
  type: SET_DEFAULT_APPLICATION_LOADING,
  payload: true,
});

export const stopDefaultApplicationLoading = () => ({
  type: SET_DEFAULT_APPLICATION_LOADING,
  payload: false,
});

export const addDefaultApplicationsList = (data) => ({
  type: GET_DEFAULT_APPLICATIONS,
  payload: data,
});
