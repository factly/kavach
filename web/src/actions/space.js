import axios from 'axios';
import {
  SET_SPACES_LOADING,
  SPACES_API,
  STOP_SPACES_LOADING,
  SET_SELECTED_APP,
  ADD_SPACES,
  ADD_SPACE,
  ADD_SPACE_TOKEN_IDS,
} from '../constants/space';
import { ORGANISATIONS_API } from '../constants/organisations';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { buildObjectOfItems, deleteKeys, getIds, getValues } from '../utils/objects';
import { addUsersList } from './users';
import { addMediaList } from './media';
import { addSpaceIDs } from './application';
import { addSpaceRoles } from './roles';
import { addSpacePolicy } from './policy';

export const createSpace = (data, id) => {
  return (dispatch, getState) => {
    dispatch(loadingSpaces());
    return axios
      .post(
        `${ORGANISATIONS_API}/${getState().organisations.selected}/applications/${id}${SPACES_API}`,
        data,
      ) // eslint-disable-next-line
      .then(() => {
        dispatch(addSuccessNotification('Space added successfully'));
        dispatch(stopLoadingSpaces());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
        dispatch(stopLoadingSpaces());
      });
  };
};

export const getSpaces = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingSpaces());
    return axios
      .get(
        `${ORGANISATIONS_API}/${getState().organisations.selected}/applications/${id}${SPACES_API}`,
      ) // eslint-disable-next-line
      .then((response) => {
        const spaceIDs = getIds(response.data);
        dispatch(addSpaceIDs(id, spaceIDs));
        deleteKeys(response.data, ['application']);
        dispatch(addSpaces(response.data));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .then(() => {
        dispatch(stopLoadingSpaces());
      });
  };
};

export const editSpace = (id, appID, data) => {
  return (dispatch, getState) => {
    dispatch(loadingSpaces());
    return axios
      .put(
        `${ORGANISATIONS_API}/${
          getState().organisations.selected
        }/applications/${appID}${SPACES_API}/${id}`,
        data,
      )
      .then(() => {
        dispatch(addSuccessNotification('Space updated successfully'));
        dispatch(stopLoadingSpaces());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
        dispatch(stopLoadingSpaces());
      });
  };
};

export const deleteSpace = (appID, id) => {
  return (dispatch, getState) => {
    dispatch(loadingSpaces());
    return axios
      .delete(
        `${ORGANISATIONS_API}/${
          getState().organisations.selected
        }/applications/${appID}${SPACES_API}/${id}`,
      )
      .then(() => {
        dispatch(addSuccessNotification('Space deleted successfully'));
        dispatch(stopLoadingSpaces());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
        dispatch(stopLoadingSpaces());
      });
  };
};

export const loadingSpaces = () => {
  return {
    type: SET_SPACES_LOADING,
    payload: true,
  };
};

export const stopLoadingSpaces = () => {
  return {
    type: STOP_SPACES_LOADING,
    payload: false,
  };
};

export const setSelectedApp = (data) => {
  return {
    type: SET_SELECTED_APP,
    payload: data,
  };
};

export const getSpaceByID = (appID, spaceID) => {
  return (dispatch, getState) => {
    dispatch(loadingSpaces());
    return axios
      .get(
        `${ORGANISATIONS_API}/${
          getState().organisations.selected
        }/applications/${appID}${SPACES_API}/${spaceID}`,
      ) // eslint-disable-next-line
      .then((response) => {
        deleteKeys([response.data], ['application']);
        dispatch(addSpaceRoles(spaceID, buildObjectOfItems(response.data.roles)));
        dispatch(addSpacePolicy(spaceID, buildObjectOfItems(response.data.policies)));
        response.data.roleIDs = getIds(response.data.roles);
        response.data.policyIDs = getIds(response.data.policies);
        delete response.data.roles;
        delete response.data.policies;
        dispatch(addSpaces([response.data]));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopLoadingSpaces());
      });
  };
};

export const addSpaces = (data) => (dispatch) => {
  const media = getValues(data, ['logo', 'logo_mobile', 'fav_icon', 'mobile_icon']);
  dispatch(addMediaList(media));
  deleteKeys(data, ['logo', 'logo_mobile', 'fav_icon', 'mobile_icon', 'application']);
  const users = getValues(data, ['users']);
  dispatch(addUsersList(users));
  data.forEach((space) => {
    space.users = getIds(space.users);
    space.tokens = space.tokens?.length ? getIds(space.tokens) : [];
  });
  dispatch({
    type: ADD_SPACES,
    payload: buildObjectOfItems(data),
  });
};

export const addSpace = (data) => {
  return {
    type: ADD_SPACE,
    payload: data,
  };
};

export const addSpaceTokenIDs = (spaceID, data) => {
  return {
    type: ADD_SPACE_TOKEN_IDS,
    payload: {
      spaceID: spaceID,
      data: data,
    },
  };
};
