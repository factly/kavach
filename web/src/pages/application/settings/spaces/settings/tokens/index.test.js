import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import '../../../../../../matchMedia.mock';
import SpaceTokens from './index';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn(() => {
    return {
      appID: 1,
    };
  }),
}));

import { getSpaceByID } from '../../../../../../actions/space';

jest.mock('../../../../../../actions/space', () => ({
  getSpaceByID: jest.fn(),
}));

let state = {
  organisations: {
    selected: 1,
  },
  profile: {
    loading: false,
    roles: {
      1: 'owner',
    },
  },
  spaces: {
    ids: [1, 2],
    loading: false,
    details: {
      1: {
        id: 1,
        name: 'Test Space 1',
        description: 'Test Space 1 Description',
        application_id: 1,
      },
    },
  },
};

jest.mock('./components/TokenList', () => {
  return jest.fn((props) => <div id="mockTokenList" {...props} />);
});

let store;
const mockedDispatch = jest.fn();
mockedDispatch.mockReturnValue(Promise.resolve());
useDispatch.mockReturnValue(mockedDispatch);
store = mockStore({});
const push = jest.fn();
useHistory.mockReturnValue({ push });

describe('Space tokens Index component', () => {
  it('should render component when loading spaces', () => {
    state.spaces.loading = true;
    store = mockStore(state);
    const tree = mount(
      <Provider store={store}>
        <Router>
          <SpaceTokens />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
    expect(tree.find('Skeleton').length).toBe(1);
  });
  it('should render component when loading roles', () => {
    state.spaces.loading = false;
    state.profile.loading = true;
    store = mockStore(state);
    const tree = mount(
      <Provider store={store}>
        <Router>
          <SpaceTokens />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
    expect(tree.find('Skeleton').length).toBe(1);
  });

  it('should render component when role is not owner', () => {
    state.profile.loading = false;
    state.profile.roles = {
      1: 'user',
    };
    store = mockStore(state);
    const tree = mount(
      <Provider store={store}>
        <Router>
          <SpaceTokens />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
    expect(tree.find('Button').length).toBe(1);
  });
  it('should render component when role is owner', () => {
    state.profile.loading = false;
    state.profile.roles = {
      1: 'owner',
    };
    store = mockStore(state);
    const tree = mount(
      <Provider store={store}>
        <Router>
          <SpaceTokens />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
    expect(tree.find('Button').length).toBe(2);
    expect(tree.find('Button').at(1).text()).toBe(' Generate new tokens ');
  });
});
