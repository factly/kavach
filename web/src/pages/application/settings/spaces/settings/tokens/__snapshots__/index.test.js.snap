// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space tokens Index component should render component when loading roles 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceTokens>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/undefined/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/undefined/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/undefined/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
             Tokens in 
          </h2>
          <Skeleton>
            <div
              className="ant-skeleton"
            >
              <div
                className="ant-skeleton-content"
              >
                <Title
                  prefixCls="ant-skeleton-title"
                  width="38%"
                >
                  <h3
                    className="ant-skeleton-title"
                    style={
                      Object {
                        "width": "38%",
                      }
                    }
                  />
                </Title>
                <Paragraph
                  prefixCls="ant-skeleton-paragraph"
                  rows={3}
                  width="61%"
                >
                  <ul
                    className="ant-skeleton-paragraph"
                  >
                    <li
                      key="0"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="1"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="2"
                      style={
                        Object {
                          "width": "61%",
                        }
                      }
                    />
                  </ul>
                </Paragraph>
              </div>
            </div>
          </Skeleton>
        </div>
      </SpaceTokens>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space tokens Index component should render component when loading spaces 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceTokens>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/undefined/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/undefined/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/undefined/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
             Tokens in 
          </h2>
          <Skeleton>
            <div
              className="ant-skeleton"
            >
              <div
                className="ant-skeleton-content"
              >
                <Title
                  prefixCls="ant-skeleton-title"
                  width="38%"
                >
                  <h3
                    className="ant-skeleton-title"
                    style={
                      Object {
                        "width": "38%",
                      }
                    }
                  />
                </Title>
                <Paragraph
                  prefixCls="ant-skeleton-paragraph"
                  rows={3}
                  width="61%"
                >
                  <ul
                    className="ant-skeleton-paragraph"
                  >
                    <li
                      key="0"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="1"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="2"
                      style={
                        Object {
                          "width": "61%",
                        }
                      }
                    />
                  </ul>
                </Paragraph>
              </div>
            </div>
          </Skeleton>
        </div>
      </SpaceTokens>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space tokens Index component should render component when role is not owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceTokens>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/undefined/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/undefined/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/undefined/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
             Tokens in 
          </h2>
          <Space
            direction="vertical"
          >
            <div
              className="ant-space ant-space-vertical"
              style={Object {}}
            >
              <Item
                className="ant-space-item"
                direction="vertical"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                wrap={false}
              />
              <Item
                className="ant-space-item"
                direction="vertical"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <mockConstructor
                    appID={1}
                    role="user"
                  >
                    <div
                      appID={1}
                      id="mockTokenList"
                      role="user"
                    />
                  </mockConstructor>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </SpaceTokens>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space tokens Index component should render component when role is owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceTokens>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/undefined/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/undefined/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/undefined/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
             Tokens in 
          </h2>
          <Space
            direction="vertical"
          >
            <div
              className="ant-space ant-space-vertical"
              style={Object {}}
            >
              <Item
                className="ant-space-item"
                direction="vertical"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginBottom": 8,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "display": "flex",
                        "justifyContent": "end",
                      }
                    }
                  >
                    <Link
                      key="1"
                      to={
                        Object {
                          "pathname": "/applications/1/settings/spaces/undefined/settings/tokens/create",
                        }
                      }
                    >
                      <LinkAnchor
                        href="/applications/1/settings/spaces/undefined/settings/tokens/create"
                        navigate={[Function]}
                      >
                        <a
                          href="/applications/1/settings/spaces/undefined/settings/tokens/create"
                          onClick={[Function]}
                        >
                          <Button
                            type="primary"
                          >
                            <Wave
                              disabled={false}
                            >
                              <button
                                className="ant-btn ant-btn-primary"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                <LoadingIcon
                                  existIcon={false}
                                  loading={false}
                                  prefixCls="ant-btn"
                                >
                                  <CSSMotion
                                    motionName="ant-btn-loading-icon-motion"
                                    onAppearActive={[Function]}
                                    onAppearStart={[Function]}
                                    onEnterActive={[Function]}
                                    onEnterStart={[Function]}
                                    onLeaveActive={[Function]}
                                    onLeaveStart={[Function]}
                                    removeOnLeave={true}
                                    visible={false}
                                  >
                                    <DomWrapper />
                                  </CSSMotion>
                                </LoadingIcon>
                                <span
                                  key=".0"
                                >
                                   Generate new tokens 
                                </span>
                              </button>
                            </Wave>
                          </Button>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                </div>
              </Item>
              <Item
                className="ant-space-item"
                direction="vertical"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <mockConstructor
                    appID={1}
                    role="owner"
                  >
                    <div
                      appID={1}
                      id="mockTokenList"
                      role="owner"
                    />
                  </mockConstructor>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </SpaceTokens>
    </Router>
  </BrowserRouter>
</Provider>
`;
